messaging/coap/observe.c

```diff
@@ -433,172 +435,196 @@ coap_remove_observer_by_resource(const oc_resource_t *rsc)
 /*- Notification ------------------------------------------------------------*/
 /*---------------------------------------------------------------------------*/
 
+static void
+send_notification_separate_response(const coap_observer_t *obs,
+                                    oc_response_t *response,
+                                    const oc_string_t *uri)
+{
+  coap_packet_t req;
+#ifdef OC_TCP
+  if (obs->endpoint.flags & TCP) {
+    coap_tcp_init_message(&req, COAP_GET);
+  } else
+#endif /* OC_TCP */
+  {
+    coap_udp_init_message(&req, COAP_TYPE_NON, COAP_GET, 0);
+  }
+  memcpy(&req.token, obs->token, obs->token_len);
+  req.token_len = obs->token_len;
+
+  coap_set_header_uri_path(&req, oc_string(*uri), oc_string_len(*uri));
+
+  OC_DBG("send_notification: creating separate response for "
+         "notification");
+#ifdef OC_BLOCK_WISE
+  uint16_t block2_size = obs->block2_size;
+#else  /* !OC_BLOCK_WISE */
+  uint16_t block2_size = 0;
+#endif /* OC_BLOCK_WISE */
+  if (coap_separate_accept(&req, response->separate_response, &obs->endpoint,
+                           obs->obs_counter, block2_size)) {
+    response->separate_response->active = 1;
+  }
+}
+
+#ifdef OC_BLOCK_WISE
+
+static int
+send_notification_blockwise(coap_packet_t *notification, coap_observer_t *obs,
```

Comment:
send_notification_blockwise -> prepare_notification_blockwise

it doesn't send anything